test/scUSDS.t.sol

```diff
@@ -0,0 +1,162 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity ^0.8.19;
+
+import "forge-std/console2.sol";
+import "forge-std/Test.sol";
+import {ERC20} from "solmate/tokens/ERC20.sol";
+import {WETH} from "solmate/tokens/WETH.sol";
+import {FixedPointMathLib} from "solmate/utils/FixedPointMathLib.sol";
+import {SafeTransferLib} from "solmate/utils/SafeTransferLib.sol";
+import {IPool} from "aave-v3/interfaces/IPool.sol";
+import {IAToken} from "aave-v3/interfaces/IAToken.sol";
+import {IVariableDebtToken} from "aave-v3/interfaces/IVariableDebtToken.sol";
+import {IPoolDataProvider} from "aave-v3/interfaces/IPoolDataProvider.sol";
+import {ERC4626} from "solmate/mixins/ERC4626.sol";
+
+import {Constants as C} from "../src/lib/Constants.sol";
+import {ILendingPool} from "../src/interfaces/aave-v2/ILendingPool.sol";
+import {IProtocolDataProvider} from "../src/interfaces/aave-v2/IProtocolDataProvider.sol";
+import {IAdapter} from "../src/steth/IAdapter.sol";
+import {AaveV3ScUsdtAdapter} from "../src/steth/scUsdt-adapters/AaveV3ScUsdtAdapter.sol";
+import {UsdtWethPriceConverter} from "../src/steth/priceConverter/UsdtWethPriceConverter.sol";
+
+import {scWETH} from "../src/steth/scWETH.sol";
+import {scCrossAssetYieldVault} from "../src/steth/scCrossAssetYieldVault.sol";
+import {ISwapRouter} from "../src/interfaces/uniswap/ISwapRouter.sol";
+import {AggregatorV3Interface} from "../src/interfaces/chainlink/AggregatorV3Interface.sol";
+import {PriceConverter} from "../src/steth/priceConverter/PriceConverter.sol";
+import {ISinglePairPriceConverter} from "../src/steth/priceConverter/ISinglePairPriceConverter.sol";
+import {ISinglePairSwapper} from "../src/steth/swapper/ISinglePairSwapper.sol";
+import {IVault} from "../src/interfaces/balancer/IVault.sol";
+import {ICurvePool} from "../src/interfaces/curve/ICurvePool.sol";
+import {ILido} from "../src/interfaces/lido/ILido.sol";
+import {IwstETH} from "../src/interfaces/lido/IwstETH.sol";
+import "../src/errors/scErrors.sol";
+import {Address} from "openzeppelin-contracts/utils/Address.sol";
+import {MainnetAddresses as M} from "../script/base/MainnetAddresses.sol";
+import {UsdtWethSwapper} from "../src/steth/swapper/UsdtWethSwapper.sol";
+import {scUSDS} from "../src/steth/scUSDS.sol";
+import {IDaiUsds} from "../src/interfaces/sky/IDaiUsds.sol";
+import {scSDAI} from "../src/steth/scSDAI.sol";
+import {scDAI} from "../src/steth/scDAI.sol";
+import {SDaiWethSwapper} from "../src/steth/swapper/SDaiWethSwapper.sol";
+import {SDaiWethPriceConverter} from "../src/steth/priceConverter/SDaiWethPriceConverter.sol";
+import {SparkScSDaiAdapter} from "../src/steth/scSDai-adapters/SparkScSDaiAdapter.sol";
+
+contract scUSDSTest is Test {
```

Comment:
add one test case that will confirm that we are actually withdrawing more than initial deposit value when the vault is making profits